mailpoet/lib/Subscription/Manage.php

```diff
@@ -153,7 +153,7 @@ private function updateSubscriptions(SubscriberEntity $subscriber, array $subscr
       if ($allowedSegments && !in_array($segment->getId(), $allowedSegments)) {
         continue;
       }
-      if (!in_array($subscriberSegment->getId(), $segmentsIds)) {
+      if (!in_array($segment->getId(), $segmentsIds)) {
```

Comment:
Was wondering if we could adjust the existing tests to catch such and issue, what do you think?